### PR TITLE
Metal backend: linear with bias decompose pass

### PR DESCRIPTION
**Pull request by manuelcandales**
This pull request introduces a new Metal-specific graph transformation pass for the Apple Metal backend, aimed at decomposing `aten.linear` operations into simpler operations to improve compatibility and avoid reliance on `addmm`. It also adds the new pass to the backend's custom passes, exposes it in the package, and introduces a corresponding test module.

**Metal Backend Linear Decomposition Pass:**

* Added a new `DecomposeLinearPass` in `decompose_linear_pass.py`, which rewrites `aten.linear` nodes in the computation graph into a sequence of `matmul` and `add` operations. For 2D inputs, it temporarily unsqueezes to 3D to avoid triggering `addmm`, then squeezes back to 2D, ensuring compatibility with Metal's capabilities.
* Registered the new `DecomposeLinearPass` in the Metal backend's `get_custom_passes` method, so it is applied during compilation.
* Exposed `DecomposeLinearPass` in the `passes` module's `__init__.py` for easier imports and future extensibility.

**Testing and Module Registry:**

* Added a new `LinearWithBias` module to the module registry in the test suite, providing a model that exercises the new pass and ensuring coverage for linear layers with bias.

### CODE REVIEW

**Review by mergennachin — APPROVED** [review]

File: backends/apple/metal/tests/test_modules.py [hunk 1]
```diff
@@ -208,6 +208,23 @@ def forward(self, x: torch.Tensor):
...
+        super().__init__()
+        self.linear = nn.Linear(7, 101, bias=True)
+
+    def forward(self, x: torch.Tensor):
+        return self.linear(x)
+
+
+MODULE_REGISTRY["linear_bias"] = {
+    "model_class": LinearWithBias,
+    "input_shapes": [(127, 7)],
```
Comment (digantdesai) [inline backends/apple/metal/tests/test_modules.py:223]: Add more tests?
- higher than 2 rank input
- dynamic shape input for batch dim
- w/ an actual add instead of the bias
